src/main/python/trec-covid/filter_run_with_qrels.py

```diff
@@ -0,0 +1,95 @@
+#
+# Pyserini: Python interface to the Anserini IR toolkit built on Lucene
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Filter a TREC run file to either retain only docids that are judged (--retain) or throw away documents that are
+judged (--discard)."""
+
+import argparse
+import sys
+from collections import defaultdict
+
+
+def load_qrels(qrels):
+    judged_docids = defaultdict(set)
+    with open(qrels) as f:
+        for line in f:
+            cols = line.split()
+            qid = cols[0]
+            docid = cols[2]
+            judged_docids[qid].add(docid)
+    return judged_docids
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=lambda prog: argparse.HelpFormatter(prog, width=100))
+    parser.add_argument('--qrels', type=str, metavar='qrels', help='Qrels.', required=True)
+    parser.add_argument('--input', type=str, metavar='run', help='Input run.', required=True)
+    parser.add_argument('--output', type=str, metavar='run', help='Output run.', required=True)
+    parser.add_argument('--runtag', type=str, default=None, metavar='runtag', help='Runtag.')
+    parser.add_argument('--k', type=int, metavar='hits', default=1000, help='Number of hits to retain per topic.')
+    parser.add_argument('--retain',  action='store_true',
+                        help="Retain judged docids, i.e., throw away all unjudged documents.")
+    parser.add_argument('--discard',  action='store_true',
+                        help="Discard judged docids, i.e., keep only documents that are unjudged.")
+    args = parser.parse_args()
+
+    if (args.retain and args.discard) or (not args.retain and not args.discard):
```

Comment:
not a big thing, but `assert` is preferred as it does not need `sys.exit()`

Reply:
Since this is an actual command-line program, `assert` will throw stack trace which I think is actually more confusing.

Reply:
Will change to just `return`,